drivers/axi_core/axi_dmac/axi_dmac.c

```diff
@@ -410,18 +410,20 @@ int32_t axi_dmac_transfer_start(struct axi_dmac *dmac,
 		}
 	}
 
-	/* Cyclic transfers set to HW or SW depending on size for MEM to DEV. */
+	/* Clear the DMA_CYCLIC flag for all transfers */
+	axi_dmac_read(dmac, AXI_DMAC_REG_FLAGS, &reg_val);
+	reg_val = reg_val & ~DMA_CYCLIC;
+	axi_dmac_write(dmac, AXI_DMAC_REG_FLAGS, reg_val);
+
+	/* Cyclic transfers set to HW for MEM to DEV if smaller than maximum transfer size
+	 * and DMA has this feature. */
 	if ((dmac->direction == DMA_MEM_TO_DEV) && (dmac->transfer.cyclic == CYCLIC)) {
```

Comment:
combine these conditions with && instead of nested ifs

Reply:
Will do this.